app/Models/User.php

```diff
@@ -541,11 +558,16 @@ public function scopeAccessInactive(Builder $query): Builder
     }
 
     /**
-     * Scope a query to automatically include only those eligible to be granted BuzzCard access
+     * Scope a query to automatically include only those eligible to be granted BuzzCard access.
      */
     public function scopeBuzzCardAccessEligible(Builder $query): Builder
     {
-        return $query->accessActive()->where('buzzcard_access_opt_out', false);
+        return $query->accessActive()
+            ->where('buzzcard_access_opt_out', false)
+            ->whereIn('primary_affiliation', ['student', 'faculty', 'staff'])
+            ->whereDoesntHave('duesPackages', static function (Builder $q): void {
+                $q->where('restricted_to_students', false);
```

Comment:
This should have to be paid.

Reply:
This is handled by the `accessActive()` scope which checks for either a currently active and paid dues transaction or an access override.